src/Initializer.js

```diff
@@ -1,289 +1,407 @@
 'use strict'
 
+var beautify = require('js-beautify').js_beautify
 /**
  * Initializer
  */
 class Initializer {
 
   /**
-   * Compile (static)
+   * constructor
    */
-  static compile (schema) {
-    let initializer = new Initializer(schema)
-    initializer.parse()
-    return initializer.compile()
+  constructor (schema, options) {
+    Object.assign(this, options || {})
+    this.root = this.root || this
+
+    this.root.depth = this.root.depth || 1
+
+    if (this.level > this.root.depth) {
+      this.root.depth = this.level
+    }
+
+    this.level = this.level || 0
+    this.schema = schema
   }
 
   /**
-   * Constructor
+   * compile (static)
    */
-  constructor (schema) {
-    this.schema = schema
+  static compile (schema) {
+    let initializer = new Initializer(schema)
+    let block = initializer.compile()
+
+    //console.log(beautify(block))
+    try {
+      return new Function('target', 'source', 'options', block)
+    } catch (e) {
+      console.log(e, e.stack)
+    }
   }
 
   /**
-   * Parse
+   * compile
    */
-  parse () {
-    let schema = this.schema
-    let operations = this.operations = new Map()
+  compile () {
+    let { root, depth, level } = this
+    let declarations = ``
```

Comment:
Any specific reason for backticks instead of single quotes, for these string inits?